File: server/BeerWorkshop.Application/Dto/Responses/WriteOff/ExpiringProductInventoryItemResponseDto.cs
using BeerWorkshop.Database.Enums;

namespace BeerWorkshop.Application.Dto.Responses.WriteOff;

public record ExpiringProductInventoryItemResponseDto(
    Guid Id,
    decimal Quantity,
    DateTime IncomingDate,
    DateTime? OpeningDate,
    DateTime ManufactureDate,
    int ExpirationTime,
    ExpirationMeasure ExpirationMeasure,
    ExpirationCountingDateType ExpirationCountingDateType
);

File: client/src/redux/dtos/responses/writeOff/expiringProductInventoryItemResponseDto.ts
import {
  ExpirationCountingDateType,
  ExpirationMeasureType,
} from "../../../enums";

export interface ExpiringProductInventoryItemResponseDto {
  id: string;
  quantity: number;
  incomingDate: Date;
  manufactureDate: Date;
  openingDate?: Date;
  expirationTime: number;
  expirationMeasure: ExpirationMeasureType;
  expirationCountingDateType: ExpirationCountingDateType;
}
